cypress/e2e/awx/resources/inventorySource.cy.ts

```diff
@@ -70,7 +77,17 @@ describe('Inventory source page', () => {
     cy.selectDropdownOptionByResourceName('source_control_type', 'Sourced from a Project');
     cy.selectDropdownOptionByResourceName('project', project.name);
     cy.selectDropdownOptionByResourceName('inventory', 'Dockerfile');
-    cy.getBy('.pf-v5-c-input-group > .pf-v5-c-button').click();
+    cy.getBy('[data-cy="execution-environment-select-form-group"]').within(() => {
+      cy.getBy('.pf-v5-c-input-group > .pf-v5-c-button').click();
+    });
+    cy.getBy('[data-ouia-component-type="PF5/ModalContent"]').within(() => {
+      cy.searchAndDisplayResource(executionEnvironmentName);
+      cy.getBy('[data-cy="checkbox-column-cell"] > label').click();
+      cy.contains('button', 'Confirm').click();
+    });
+    cy.getBy('[data-cy="credential-select-form-group"]').within(() => {
+      cy.getBy('.pf-v5-c-input-group > .pf-v5-c-button').click();
+    });
     cy.getBy('[data-ouia-component-type="PF5/ModalContent"]').within(() => {
```

Comment:
I think this is not part of your work, but we should try to find a different selector here too.